File: backend/app/schemas/style.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class StyleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    prompt: str
    reference_image_url: Optional[str] = None
    creator_id: int
    created_at: datetime
    updated_at: datetime

File: src/types/style.ts
export interface Style {
  id: number;
  name: string;
  prompt: string;
  reference_image_url?: string;
  creator_id: number;
  created_at: string;
  updated_at: string;
}

export interface CreateStyleRequest {
  name: string;
  prompt: string;
}
